addons/binding/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/handler/ExecHandler.java

```diff
@@ -0,0 +1,263 @@
+    }
+
+    @Override
+    public void dispose() {
+        if (executionJob != null && !executionJob.isCancelled()) {
+            executionJob.cancel(true);
+            executionJob = null;
+        }
+    }
+
+    protected Runnable periodicExecutionRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+
+            String commandLine = (String) getConfig().get(COMMAND);
+
+            int timeOut = 60000;
+            if (((BigDecimal) getConfig().get(TIME_OUT)) != null) {
+                timeOut = ((BigDecimal) getConfig().get(TIME_OUT)).intValue() * 1000;
+            }
+
+            if (commandLine != null && !commandLine.isEmpty()) {
+
+                updateState(new ChannelUID(getThing().getUID(), ExecBindingConstants.RUNNING), OnOffType.ON);
+
+                // For some obscure reason, when using Apache Common Exec, or using a straight implementation of
+                // Runtime.Exec(), on Mac OS X (Yosemite and El Capitan), there seems to be a lock race condition
+                // randomly appearing (on UNIXProcess) *when* one tries to gobble up the stdout and sterr output of the
+                // subprocess in separate threads. It seems to be common "wisdom" to do that in separate threads, but
+                // only when keeping everything between .exec() and .waitfor() in the same thread, this lock race
+                // condition seems to go away. This approach of not reading the outputs in separate threads *might* be a
+                // problem for external commands that generate a lot of output, but this will be dependent on the limits
+                // of the underlying operating system.
+
+                Process proc = null;
+                try {
+                    proc = rt.exec(commandLine.toString());
+                } catch (Exception e) {
+                    logger.error("An exception occured while executing '{}' : '{}'",
+                            new Object[] { commandLine.toString(), e.getMessage() });
+                    updateState(new ChannelUID(getThing().getUID(), ExecBindingConstants.RUNNING), OnOffType.OFF);
+                    updateState(new ChannelUID(getThing().getUID(), ExecBindingConstants.OUTPUT),
+                            new StringType(e.getMessage()));
+                    return;
+                }
+
+                StringBuilder outputBuilder = new StringBuilder();
+                StringBuilder errorBuilder = new StringBuilder();
+
+                try {
+                    InputStreamReader isr = new InputStreamReader(proc.getInputStream());
+                    BufferedReader br = new BufferedReader(isr);
+                    String line = null;
+                    while ((line = br.readLine()) != null) {
+                        outputBuilder.append(line).append("\n");
+                        logger.debug("Exec [{}]: '{}'", "OUTPUT", line);
+                    }
+                    br.close();
```

Comment:
you should do the closing in the finally block or use try-with-resources instead.